### ISSUE TITLE
Skia performance comparison 

### ISSUE THREAD

**Issue by LifeIsStrange** [blend2d/blend2d#36]
I find this project fascinating, and has a lot of room for improvement.
BUT https://blend2d.com/performance.html
Should show a comparison against latest Skia.

Less importantly, it would be nice to compare performance with pathfinder and other state of the art renderers
https://github.com/pcwalton/pathfinder
https://github.com/intel/fastuidraw
https://www.google.fr/amp/s/www.amanithvg.com/

I can't find on internet an exhaustive performance comparison of 2D renderers. 
Your website could become a great reference for choice.
Thus it would be nice to compare performance with direct2D and openVG too :) and even higher level libs like SDL?

**Comment by kobalicek** [comment]
A possible duplicate of #3 where most questions should have been already answered?

Anyone is welcome to contribute SKIA support to (blend2d-bench)[https://github.com/blend2d/blend2d-bench] or implement a better benchmarking tool and to objectively present possible findings. `blend2d-bench` tool was written mainly for development and the tests were designed to exercise Blend2D pipelines.

I think comparing with GPU implementations would make sense after multithreaded rendere lands as GPU implementations use the whole GPU hardware whereas the current software-based renderer only uses a single core.

**Comment by LifeIsStrange** [comment]
@kobalicek thank you for the fast answer.
This issue is indeed a duplicate, I just wanted to bump up the priority of adding skia. 
I agree that comparing blend2d to skia on openGL/Vulkan would be unfair.
But skia already has, I believe an optional, cpu only backend. (still unfair because multithreaded but less unfair)
I'm really interested to see the eventual improvements that multithreading will bring.
Btw modern OpenMP allow to very easily have code that is multithreaded, vectorized (SIMD) and even offloaded to the gpu ! With just a switch.

I agree too on openVG, but I have hopes for NV_path.
Anyway, I guess it would benefit you to show a performance comparison of blend2d vs skia on cpu backend, because if it would outperform or have similar performance despite the fact that skia is 40000 commits of an army of experts, would prove that you have a very innovative, almost revolutionary optimization.
And thus, Google would probably hire or fund you to either bring blend2d optimization on skia or to help them switch from skia to blend2d.

**Comment by kobalicek** [comment]
> I just wanted to bump up the priority of adding skia

You are welcome! SKIA comparison would be really nice to have, but I was not lucky to build it in the past and after that I just gave up and made it low priority. So maybe when they add support for CMake I would give it another shot - or if somebody else is willing to invest time in this?

> I agree too on openVG, but I have hopes for NV_path

NV_path - I'm not a big fan of it. I think that in general such technology should work on all GPUs and not be restricted to a single vendor with binary only drivers. I have actually tried it, downloaded the SDK and then noticed only half of the samples actually compiled on my Linux machine. I have GTX 1070 GPU and I have noticed that NV_path quality was much worse than I initially expected - it was definitely not 256 shades of gray, I would say it was close to 8x or 16x.

> because if it would outperform or have similar performance despite the fact that skia is 40000 commits of an army of experts, would prove that you have a very innovative, almost revolutionary optimization.

I would humbly say that Blend2D is just few really optimized parts connected together that as a whole form the library. SKIA has much more features that we don't even plan to have or that would take a lifetime to do.

> And thus, Google would probably hire or fund you to either bring blend2d optimization on skia or to help them switch from skia to blend2d.

SKIA devs know about Blend2D so I'm sure they would let me know if they are interested in funding the project or anything else. Maybe when Blend2D gets multithreaded rasterizer we would talk again, let's see.
